File: examples/app-model/BooksAndRecords/src/main/java/com/ms/booksandrecords/controller/DealController.java
package com.ms.booksandrecords.controller;

import com.ms.booksandrecords.commands.DealCmd;
import com.ms.booksandrecords.service.DealEvent;
import com.ms.booksandrecords.service.ServiceStatefulApp;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class DealController {

    @Autowired
    ServiceStatefulApp serviceStatefulApp;

    @PostMapping(value="/deal", consumes = MediaType.APPLICATION_JSON_VALUE, produces = "application/json")
    public DealEvent processDealCmd(@RequestBody DealCmd dealCmd){
        return serviceStatefulApp.logic(dealCmd);
    }
}

File: examples/workspaces/booksandrecords-spring-boot/src/main/scala/com/ms/booksandrecords/controller/DealController.scala
package com.ms.booksandrecords.controller

import com.ms.booksandrecords.commands.DealCmd
import com.ms.booksandrecords.service.{DealEvent, ServiceStatefulApp}
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.{PostMapping, RequestBody, RestController}

@RestController class DealController {

  @Autowired private val serviceStatefulApp: ServiceStatefulApp = null

  @PostMapping(value = Array("/deal"), consumes = Array(MediaType.APPLICATION_JSON_VALUE), produces = Array("application/json"))
  def processDealCmd(@RequestBody dealCmd: DealCmd): DealEvent = {
    serviceStatefulApp.logic(dealCmd)
  }
}